File: app/src/main/java/com/coppermobile/mysamplemvvmdatabindinglivedata/adapters/CommentsAdapter.java
package com.coppermobile.mysamplemvvmdatabindinglivedata.adapters;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import com.coppermobile.mysamplemvvmdatabindinglivedata.R;
import com.coppermobile.mysamplemvvmdatabindinglivedata.data.Comments;
import com.coppermobile.mysamplemvvmdatabindinglivedata.databinding.ListCommentsBinding;
import com.coppermobile.mysamplemvvmdatabindinglivedata.utils.IShowPopUpListener;

import java.util.List;

public class CommentsAdapter extends RecyclerView.Adapter<CommentsAdapter.MyViewHolder> {

    private List<Comments> commentsList;
    private LayoutInflater layoutInflater;
    private IShowPopUpListener iShowPopUpListener;

    public CommentsAdapter(IShowPopUpListener iShowPopUpListener) {
        this.iShowPopUpListener = iShowPopUpListener;
    }

    public void addData(List<Comments> commentsList) {
        this.commentsList = commentsList;
        notifyDataSetChanged();
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, int i) {
        if (layoutInflater == null) {
            layoutInflater = LayoutInflater.from(viewGroup.getContext());
        }
        ListCommentsBinding binding;
        binding = DataBindingUtil.inflate(layoutInflater, R.layout.list_comments, viewGroup, false);
        return new MyViewHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder myViewHolder, int i) {
        Comments comments = commentsList.get(i);
        if (comments != null) {
            myViewHolder.binding.setComments(comments);
            myViewHolder.binding.setCallback(iShowPopUpListener);
        }
    }

    @Override
    public int getItemCount() {
        return commentsList != null && !commentsList.isEmpty() ? commentsList.size() : 0;
    }

    class MyViewHolder extends RecyclerView.ViewHolder {

        private ListCommentsBinding binding;

        MyViewHolder(ListCommentsBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: app/src/main/java/com/coppermobile/kotlinsamplemvvmlivedataroom/adapters/CommentsAdapter.kt
package com.coppermobile.kotlinsamplemvvmlivedataroom.adapters

import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView
import butterknife.BindView
import butterknife.ButterKnife
import com.coppermobile.kotlinsamplemvvmlivedataroom.R
import com.coppermobile.kotlinsamplemvvmlivedataroom.data.Comments
import com.coppermobile.kotlinsamplemvvmlivedataroom.utils.IShowPopUpListener

class CommentsAdapter(private val iShowPopUpListener: IShowPopUpListener?) : RecyclerView.Adapter<CommentsAdapter.MyViewHolder>() {

    private var commentsList: MutableList<Comments>? = null

    fun addComments(commentsList: MutableList<Comments>) {
        this.commentsList = commentsList
        notifyDataSetChanged()
    }

    fun addSingleComment(comments: Comments) {
        commentsList?.add(comments)
        notifyDataSetChanged()
    }

    fun deleteComment(position: Int) {
        commentsList?.removeAt(position)
        notifyDataSetChanged()
    }

    override fun onCreateViewHolder(parent: ViewGroup, p1: Int): MyViewHolder {
        return MyViewHolder(LayoutInflater.from(parent.context).inflate(R.layout.list_comments, parent, false))
    }

    override fun onBindViewHolder(holder: MyViewHolder, p1: Int) {
        val comments = commentsList?.get(p1)
        if (comments != null) {
            holder.tvComments.text = comments.comments
            holder.tvDate.text = comments.date
            holder.ivMenu.setOnClickListener { iShowPopUpListener?.showPopupMenu(comments, holder.adapterPosition, holder.ivMenu) }
        }
    }

    override fun getItemCount(): Int {
        return if (commentsList != null && !commentsList!!.isEmpty()) commentsList!!.size else 0
    }

    inner class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {

        @BindView(R.id.tv_comments)
        lateinit var tvComments: TextView
        @BindView(R.id.tv_date)
        lateinit var tvDate: TextView

        @BindView(R.id.iv_menu)
        lateinit var ivMenu: ImageView

        init {
            ButterKnife.bind(this, itemView)
        }
    }
}